### PR TITLE
MSCNG: add raw RSA support

### PR DESCRIPTION
**Pull request by vmiklos**
And a couple related smaller things. I think the dsig support in the mscng backend starts to look reasonable with this.

### CODE REVIEW

File: src/mscng/certkeys.c [hunk 1]
```diff
@@ -906,6 +911,78 @@ static void xmlSecMSCngKeyDataDsaDebugXmlDump(xmlSecKeyDataPtr data, FILE* outpu
...
+            xmlSecKeyDataGetName(data));
+        goto done;
+    }
+    hKey = 0;
+
+    /* success */
+    res = 0;
+
+done:
+    if (hAlg != 0) {
```
Comment (lsh123) [inline src/mscng/certkeys.c]: Might not be that important but... hKey was created from hAlg. I would keep the cleanup order reversed (i.e. hKey destroyed before hAlg).

File: src/mscng/certkeys.c [hunk 2]
```diff
@@ -1009,6 +1086,370 @@ static void xmlSecMSCngKeyDataRsaDebugXmlDump(xmlSecKeyDataPtr data, FILE* outpu
...
+
+    keyData = NULL;
+    res = 0;
+
+done:
+    xmlSecBnFinalize(&exponent);
+    xmlSecBnFinalize(&modulus);
+    xmlSecBufferFinalize(&blob);
+
+    if(hAlg != 0) {
```
Comment (lsh123) [inline src/mscng/certkeys.c]: Same.

File: src/mscng/certkeys.c [hunk 3]
```diff
@@ -1009,6 +1086,370 @@ static void xmlSecMSCngKeyDataRsaDebugXmlDump(xmlSecKeyDataPtr data, FILE* outpu
...
+            xmlSecKeyDataGetName(data));
+        goto done;
+    }
+    hKey = 0;
+
+    /* success */
+    res = 0;
+
+done:
+    if (hAlg != 0) {
```
Comment (lsh123) [inline src/mscng/certkeys.c]: Same

File: src/mscng/certkeys.c (hunk 1 above)
Comment (vmiklos) [reply src/mscng/certkeys.c]: Makes sense.

File: src/mscng/certkeys.c (hunk 2 above)
Comment (vmiklos) [reply src/mscng/certkeys.c]: Done.

File: src/mscng/certkeys.c (hunk 3 above)
Comment (vmiklos) [reply src/mscng/certkeys.c]: Done.

**Comment by lsh123** [comment]
Thank you!

**Pull request merged**